### ISSUE TITLE
[Accessibility]: Contributors page: Filter form controls

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#35356]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34427

### Problem description

Time period buttons aren't linked to the form label. For example, screen readers only read "Button, 7 days". They should read "Time period, Button, 7 days"
<img width="697" alt="image" src="https://user-images.githubusercontent.com/9516420/168042722-bc575c4f-0a85-44eb-94fb-09e752b405df.png">

### Expected behavior

We should attach the label to this buttons correctly. We should be able to do this by:
1. Giving the label an id
2. Using `aria-labelledby` on each button

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
